api/mindexd/main.go

```diff
@@ -0,0 +1,252 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	logging "github.com/ipfs/go-log/v2"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"github.com/textileio/go-threads/util"
+	"github.com/textileio/textile/v2/api/mindexd/service"
+	"github.com/textileio/textile/v2/cmd"
+)
+
+const daemonName = "mindexd"
+
+var (
+	log = logging.Logger(daemonName)
+
+	config = &cmd.Config{
+		Viper: viper.New(),
+		Dir:   "." + daemonName,
+		Name:  "config",
+		Flags: map[string]cmd.Flag{
+			"debug": {
+				Key:      "log.debug",
+				DefValue: false,
+			},
+			"logFile": {
+				Key:      "log.file",
+				DefValue: "", // no log file
+			},
+
+			// Addr config
+			"addrApi": {
+				Key:      "addr.api",
+				DefValue: "/ip4/127.0.0.1/tcp/20006",
```

Comment:
Looks good, but wondering if we should come up with a pattern for micro-service API endpoints? Does this need to be `10006` to match the Dockerfile?

Reply:
Yep, you're correct. As mentioned in the other comment, I'll switch to 5000.